design/secure-communication.md

```diff
@@ -0,0 +1,94 @@
+# Secure Communication for Contour and Envoy
+
+Status: Draft
+
+This document outlines a design for securing gRPC communication between Contour and Envoy, including all the xDS endpoints.
+
+## Goals
+
+- Achieve secure communication between Contour and Envoy, which will allow a split deployment.
+- Allow use of the SDS service in Contour with authentication.
+
+## Non Goals
+
+- Certificate management, including creation or rotation. These are left as an exercise for the reader at this time.
```

Comment:
s/exercise to the reader/These must be provided by the user. In the future we plan to automated this process to allow split deployments to become the default configuration.